Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -364,13 +364,9 @@ def bottle_formula(f)
 
     old_spec = f.bottle_specification
     if args.keep_old? && !old_spec.checksums.empty?
-      mismatches = [:root_url, :prefix, :cellar, :rebuild].reject do |key|
+      mismatches = [:root_url, :prefix, :rebuild].reject do |key|
         old_spec.send(key) == bottle.send(key)
       end
-      if old_spec.cellar == :any && bottle.cellar == :any_skip_relocation
```

Comment:
Instead of removing this could you would need to add more cases like this. An `:any_skip_relocation` can be labelled as `:any` or `:any` as with the default cellar (i.e. `cellar` is omitted) but you cannot go in the other direction: a default cellar bottle cannot be labelled as `:any` or `:any` as `:any_skip_relocation` or they will be installed incorrectly.

Short version: instead of removing this you want to use something like (this is untested): 
```ruby
if (old_spec.cellar == :any && bottle.cellar == :any_skip_relocation) ||
   ([HOMEBREW_DEFAULT_CELLAR, LINUXBREW_DEFAULT_CELLAR].include?(old_spec.cellar) &&
    [:any, :any_skip_relocation].include?(bottle.cellar))
  mismatches.delete(:cellar)
  bottle.cellar old_spec.cellar
end
```

Reply:
Nice, but it won't be enough as we get cases like these:
```
cellar: old: :any, new: "/home/linuxbrew/.linuxbrew/Cellar"
```
```
cellar: old: :any_skip_relocation, new: "/home/linuxbrew/.linuxbrew/Cellar"
```
```
cellar: old: "/home/linuxbrew/.linuxbrew/Cellar", new: :any_skip_relocation
```

Reply:
If you get those first two cases then the bottle needs to no longer be `cellar :any`.

Reply:
How about modifying this `if` condition to:
```ruby
if (old_spec.cellar == :any && bottle.cellar == :any_skip_relocation) || (ENV["CI"] && OS.linux?)
  mismatches.delete(:cellar)
  bottle.cellar old_spec.cellar
end
```

I mean, this issue is in fact specific to our workflow I would say, so maybe it could be solved like that?

Reply:
Sorry, I'm not willing to include changes (even for Linux) that are incorrect. In that case you're going to have bottles marked `cellar :any` on Linux that are not `cellar :any`. If you wish to manually edit files to do that: that's up to you but I don't want `brew bottle` doing it.

Reply:
Understandable, thanks for review. I will think about another solution.

Reply:
I followed your advice and force pushed the branch. It would not eliminate the problem completely but it will certainly mitigate it a bit.

Reply:
Sounds good.